pkg/api/v1alpha2/api.proto

```diff
@@ -0,0 +1,481 @@
+        repeated Operation operation = 1;
+    }
+    Operations operations = 2; /// List of Operation
+}
+
+/**
+ * GraphConfig contains a config of DAG
+ */
+ message GraphConfig {
+    int32 num_layers = 1; /// Number of layers
+    repeated int32 input_size = 2; /// Dimensions of input size
+    repeated int32 output_size = 3; /// Dimensions of output size
+}
+
+/**
+ * Config for operations in DAG
+ */
+ message Operation {
+     string operationType = 1; /// Type of operation in DAG
+      /** 
+     * List of ParameterAssignment
+     */
+     message ParameterAssignments {
+        repeated ParameterAssignment assignments = 1;
+    }
+    ParameterAssignments parameter_assignments = 2; /// List of ParameterAssignments
+}
+
+/**
+ * Spec of a Experiment. Experiment represents a single optimization run over a feasible space. 
+ * Each Experiment contains a configuration describing the feasible space, as well as a set of Trials.
+ * It is assumed that objective function f(x) does not change in the course of a Experiment.
+ */
+message ExperimentSpec {
+    repeated ParameterSpec parameters = 1; 
+    ObjectiveSpec objective = 2; 
+    AlgorithmSpec algorithm = 3;
+    string trial_template = 4;
+    int32 parallel_trial_count = 5;
+    int32 max_trial_count = 6;
+    string metrics_collector_type = 7;
+}
+
+message ExperimentStatus {
+    enum ExperimentConditionType {
+        CREATED = 0;
+        RUNNING = 1;
+        RESTARTING = 2;
+        SUCCEEDED = 3;
+        FAILED = 4;
+    }
+    string start_time = 1; /// RFC3339 format
+    string completion_time = 2; /// RFC3339 format
+    string last_reconcile_time = 3; /// RFC3339 format
+    ExperimentConditionType condition = 4;
+}
+
+message Experiment {
+    string name = 1; /// Name of Experiment. This is unique in DB.
+    ExperimentSpec experiment_spec = 2;
+    ExperimentStatus experiment_status = 3;
+}
+
+message ParameterAssignment {
+    string name = 1;
+    string value = 2;
+}
+
+message Metric {
+    string name = 1;
+    string value = 2;
+}
+
+message MetricLog {
+    message ValueLog {
+        string time_stamp = 1; /// RFC3339 format
+        string value = 2;
+    }
+    string name = 1;
+    repeated ValueLog valuelog = 2;
+}
+
+message Observation {
+    repeated Metric metrics = 1;
+}
+
+message ObservationLog {
+    repeated MetricLog metriclogs = 1;
+}
+
+message TrialSpec {
+    string experiment_name = 1;
+    repeated ParameterAssignment parameter_assignments = 2;
+    string run_spec = 3;
+}
+
+message TrialStatus {
+    enum TrialConditionType {
+        PENDING = 0;
+        RUNNING = 1;
+        COMPLETED = 2;
+        KILLED = 3;
+        FAILED = 4;
+    }
+    string start_time = 1; /// RFC3339 format
+    string completion_time = 2; /// RFC3339 format
+    string last_reconcile_time = 3; /// RFC3339 format
+    TrialConditionType condition = 4;
+    Observation observation = 5; /// The best observation in logs.
+}
+
+message Trial {
+    string id = 1;
```

Comment:
Is this ID or name(like in experiment)?

Reply:
Since the trials are generated by the controller, users won't name them.
I think `ID` is more suitable here.

Reply:
It is true. But, I feel that it might be better to keep the same terminology as in the Kubernetes config(matadata.name)

Reply:
make sense. "name" would be more user friendly.